contracts/MultisigCaller.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+error InvalidApproverCount(uint256 count, uint256 minimum);
+error RequiredApprovalsTooLow(uint256 required, uint256 minimum);
+error RequiredApprovalsExceedApprovers(uint256 required, uint256 approvers);
+error ZeroAddress();
+error MaxApproversReached(uint256 max);
+error InsufficientApprovers(uint256 required, uint256 approvers);
+error InvalidTransactionId(uint256 txId);
+error TransactionAlreadyExecuted(uint256 txId);
+error TransactionAlreadyApproved(uint256 txId, address approver);
+
+contract MultisigCaller is AccessControlEnumerable, ReentrancyGuard {
+    bytes32 public constant APPROVER_ROLE = keccak256("APPROVER_ROLE");
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    uint256 public constant MIN_APPROVERS = 2;
+    uint256 public constant MAX_APPROVERS = 10;
```

Comment:
Check public everywhere, if it is not meant to be public keep external/internal/private as needed

Reply:
Done in 5c8331ad3f4ec10d3d2183419af2965fbd38f3f4.